backend/quatrix/quatrix.go

```diff
@@ -0,0 +1,1255 @@
+				Advanced: true,
+				Default:  "4s",
+			},
+			{
+				Name:     "minimal_chunk_size",
+				Help:     "The minimal size for one chunk",
+				Advanced: true,
+				Default:  "10_000_000B",
+			},
+			{
+				Name:     "maximal_summary_chunk_size",
+				Help:     "The maximal summary for all chunks. It should not be less than 'transfers'*'minimal_chunk_size'",
+				Advanced: true,
+				Default:  "100_000_000B",
+			},
+		},
+	})
+}
+
+// Options defines the configuration for Quatrix backend
+type Options struct {
+	APIKey                  string               `config:"api_key"`
+	Host                    string               `config:"host"`
+	Enc                     encoder.MultiEncoder `config:"encoding"`
+	EffectiveUploadTime     fs.Duration          `config:"effective_upload_time"`
+	MinimalChunkSize        fs.SizeSuffix        `config:"minimal_chunk_size"`
+	MaximalSummaryChunkSize fs.SizeSuffix        `config:"maximal_summary_chunk_size"`
+}
+
+// Fs represents remote Quatrix fs
+type Fs struct {
+	name                string
+	root                string
+	description         string
+	features            *fs.Features
+	opt                 Options
+	ci                  *fs.ConfigInfo
+	srv                 *rest.Client // the connection to the quatrix server
+	pacer               *fs.Pacer    // pacer for API calls
+	dirCache            *dircache.DirCache
+	uploadMemoryManager *UploadMemoryManager
+}
+
+// Object describes a quatrix object
+type Object struct {
+	fs          *Fs
+	remote      string
+	size        int64
+	modTime     time.Time
+	id          string
+	hasMetaData bool
+	obType      string
+}
+
+// trimPath trims redundant slashes from quatrix 'url'
+func trimPath(path string) (root string) {
+	root = strings.Trim(path, "/")
+	return
+}
+
+// retryErrorCodes is a slice of error codes that we will retry
+var retryErrorCodes = []int{
+	429, // Too Many Requests.
+	500, // Internal Server Error
+	502, // Bad Gateway
+	503, // Service Unavailable
+	504, // Gateway Timeout
+	509, // Bandwidth Limit Exceeded
+}
+
+// shouldRetry returns a boolean as to whether this resp and err
+// deserve to be retried.  It returns the err as a convenience
+func shouldRetry(ctx context.Context, resp *http.Response, err error) (bool, error) {
+	if fserrors.ContextError(ctx, &err) {
+		return false, err
+	}
+
+	return fserrors.ShouldRetry(err) || fserrors.ShouldRetryHTTP(resp, retryErrorCodes), err
+}
+
+// NewFs constructs an Fs from the path, container:path
+func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, error) {
+	// Parse config into Options struct
+	opt := new(Options)
+
+	err := configstruct.Set(m, opt)
+	if err != nil {
+		return nil, err
+	}
+
+	// http client
+	client := fshttp.NewClient(ctx)
+
+	// since transport is a global variable that is initialized only once (due to sync.Once)
+	// we need to reset it to have correct transport per each client (with proper values extracted from rclone config)
```

Comment:
Why do you need one per client?

Reply:
We use different `fs.ConfigInfo.Headers` for each call to `fs.NewFs` and before this change got the same for each 🙂 
Should it be an option to Quatrix backend?

Reply:
I don't see you setting `.Headers` in this code. Is this something you are doing with command line parameters? Or maybe rc calls? It makes sense if you are setting them via the rc and in which case maybe shouldn't be caching the Transport and re-using it, or perhaps only re-using the cached transport if there isn't a custom config in the `ctx`.

If you want to set headers for this backend only you should use `f.srv.SetHeader`